src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -428,9 +455,89 @@ def _visit_neighbors(self, node: gtir.FunCall) -> DataExpr:
             memlet=dace.Memlet(data=neighbors_temp, subset=neighbor_idx),
         )
 
-        assert isinstance(node.type, gtir_ts.ListType)
         return DataExpr(neighbors_node, node.type)
 
+    def _visit_map(self, node: gtir.FunCall) -> DataExpr:
+        assert isinstance(node.type, itir_ts.ListType)
+        assert isinstance(node.type.element_type, ts.ScalarType)
+        dtype = dace_utils.as_dace_type(node.type.element_type)
+
+        input_args = []
+        for arg in node.args:
+            input_args.append(self.visit(arg))
+
+        # TODO(edopao): extract offset_dim from the input arguments
+        offset_dim = gtx_common.Dimension("", gtx_common.DimensionKind.LOCAL)
+        map_index = dace_gtir_utils.get_map_variable(offset_dim)
+        connectors = [f"__arg{i}" for i in range(len(input_args))]
+
+        input_memlets = {}
+        input_nodes = {}
+        local_size: Optional[int] = None
+        for conn, input_expr in zip(connectors, input_args, strict=True):
+            if isinstance(input_expr, MemletExpr):
+                if set(input_expr.subset.size()[:-1]) != {1}:
+                    raise ValueError(f"Invalid node {node}")
+                rstart, rstop, rstep = input_expr.subset[-1]
+                assert rstart == 0 and rstep == 1
+
+                desc = self.sdfg.arrays[input_expr.node.data]
+                view, _ = self.sdfg.add_view(
```

Comment:
I am using a view because I want to expose a 1D array, along the local dimension.

Reply:
I hope it is more clear now, after the refactoring-PR #1685. The dataflow we build in this class has some loose connections on input edges. These edges are described as set of nodes, that will have to be connected to external data source nodes passing through the map entry node. For `map_` and `neighbors` these edges will terminate on the view nodes, because it was much more complicated to terminate them directly on the map entry node.